File: src/main/java/no/novari/flyt/file/FileCacheConfiguration.java
package no.novari.flyt.file;

import no.novari.cache.FintCache;
import no.novari.cache.FintCacheManager;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.UUID;

@Configuration
public class FileCacheConfiguration {
    @Bean
    FintCache<UUID, File> fileCache(FintCacheManager fintCacheManager) {
        return fintCacheManager.createCache(
                "file",
                UUID.class,
                File.class
        );
    }
}

File: src/main/kotlin/no/novari/flyt/config/FileCacheConfiguration.kt
package no.novari.flyt.config

import no.novari.cache.FintCache
import no.novari.cache.FintCacheManager
import no.novari.flyt.files.domain.FilePayload
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.util.UUID

@Configuration
class FileCacheConfiguration {
    @Bean
    fun fileCache(fintCacheManager: FintCacheManager): FintCache<UUID, FilePayload> {
        return fintCacheManager.createCache(
            "file",
            UUID::class.java,
            FilePayload::class.java,
        )
    }
}
